test/integration/auth-methods/jwt.test.js

```diff
@@ -0,0 +1,1664 @@
+
+    // RS Algorithms with passphrase
+    RSAlgorithmsWithPassPhrase = {
+        RS256: {
+            alg: 'RS256',
+            signKey: privatekeyRSAWithPassphrase,
+            publicKey: publicKeyRSAWithPassphrase,
+            passphrase: 'test1234key'
+        },
+        RS384: {
+            alg: 'RS384',
+            signKey: privatekeyRSAWithPassphrase,
+            publicKey: publicKeyRSAWithPassphrase,
+            passphrase: 'test1234key'
+        },
+        RS512: {
+            alg: 'RS512',
+            signKey: privatekeyRSAWithPassphrase,
+            publicKey: publicKeyRSAWithPassphrase,
+            passphrase: 'test1234key'
+        }
+    },
+
+    // PS algorithms
+    PSAlgorithms = {
+        PS256: {
+            alg: 'PS256',
+            signKey: privatekeyRSA,
+            publicKey: publicKeyRSA
+        },
+        PS384: {
+            alg: 'PS384',
+            signKey: privatekeyRSA,
+            publicKey: publicKeyRSA
+        },
+        PS512: {
+            alg: 'PS512',
+            signKey: privatekeyRSA,
+            publicKey: publicKeyRSA
+        }
+    },
+
+    // ES algorithms
+    ESAlgorithms = {
+        ES256: {
+            alg: 'ES256',
+            signKey: privateKeyECDSA,
+            publicKey: publicKeyECDSA
+        },
+        ES384: {
+            alg: 'ES384',
+            signKey: privateKeyECDSA,
+            publicKey: publicKeyECDSA
+        },
+        ES512: {
+            alg: 'ES512',
+            signKey: privateKeyECDSA,
+            publicKey: publicKeyECDSA
+        }
+    },
+
+    // ES Algorithms with passphrase
+    ESAlgorithmsWithPassPhrase = {
+        ES256: {
+            alg: 'ES256',
+            signKey: privateKeyECDSAWithPassphrase,
+            publicKey: publicKeyECDSAWithPassphrase,
+            passphrase: '12345678'
+        },
+        ES384: {
+            alg: 'ES384',
+            signKey: privateKeyECDSAWithPassphrase,
+            publicKey: publicKeyECDSAWithPassphrase,
+            passphrase: '12345678'
+        },
+        ES512: {
+            alg: 'ES512',
+            signKey: privateKeyECDSAWithPassphrase,
+            publicKey: publicKeyECDSAWithPassphrase,
+            passphrase: '12345678'
+        }
+    },
+
+    // all Algorithms
+    algorithmsSupported = {
+        ...HSAlgorithms,
+        ...RSAlgorithms,
+        ...PSAlgorithms,
+        ...ESAlgorithms
+    },
+
+    // algorithms with passphrase
+    algorithmsWithPassphrase = {
+        ...RSAlgorithmsWithPassPhrase,
+        ...ESAlgorithmsWithPassPhrase
+        // verify whether RSASSA-PSS (PS) algorithm is applicable
+    },
+
+    algorithms = Object.entries(algorithmsSupported);
+
+describe('jwt auth', function () {
+    let testrun;
+
+    // with invalid algorithm - root level
+    algorithms.forEach(([key]) => {
+        const { alg, signKey } = algorithmsSupported[key];
+
+        describe(`with invalid algorithm for ${alg} algorithm`, function () {
```

Comment:
The describe block string doesn't seem to be easy to read.
```suggestion
        describe(`with invalid algorithm key with "${alg}" algorithm set in header`, function () {
```